pkg/controllers/reconciler.go

```diff
@@ -189,26 +258,33 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	ds := &childDaemonSets.Items[0]
 
+	if UpgradePlugins {
```

Comment:
would it make sense to set `func()` based on `UpgradePlugins`. This would save complexity.